xdsl/dialects/builtin.py

```diff
@@ -575,9 +575,10 @@ def from_int_list(
 
     @staticmethod
     @builder
-    def from_float_list(
-            type: VectorOrTensorOf[AnyFloat],
-            data: List[float | AnyFloatAttr]) -> DenseIntOrFPElementsAttr:
+    def create_dense_float(
+        type: VectorOrTensorOf[AnyFloat],
+        data: List[int | AnyIntegerAttr | float | AnyFloatAttr]
```

Comment:
Actually, `AnyIntegerAttr` should not be accepted here, having one value of this type would trigger an exception (trying to translate an `IntegerAttr` to a `float`).

Reply:
True, though I feel that it should be if we were to allow ìnt` here...

Reply:
I am not completely sure if we would like to do that.
I feel here, it makes sense to go from `int` to `FloatAttr` (given a type), since we are converting an arbitrary-sized integer to a float. However, a conversion from `IntegerAttr` to `FloatAttr` makes a little bit less sense in my opinion, since we are now converting something that is bounded (and might have been converted already).
So maybe it would be dangerous to do this conversion, but maybe I'm also wrong and I'm overthinking this.

Reply:
So I don't have a strong opinion, either we add the conversion, or we change the type hint.

Reply:
I am fine with the latter and potentially addressing this issue once it actually arises somewhere...